### PR TITLE
Athena Query Execution Pipeline

### PR DESCRIPTION
**Pull request by anmolsgandhi**
Issue #, if available: #17

Description of changes:
This PR provides a new pattern that uses DDK for AthenaQuertExecution Pipeline example

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by noah-paige — COMMENTED** [review]
Comments:
1. The `data` and `analytics` prefixes and `athena_data` and `athena_analytics` Glue DB names seem pretty arbitrary (could these be parametrized with defaults remaining as above?)

2. Would the `table` parameter get confusing when you are doing JOIN across multiple tables?

3. Would there ever be a need to trigger the Athena Stage as soon as new data comes in to `athena_data` rather than using cronExpressions? Is that possible to have glue data catalog changes as a listener?

**Comment by anmolsgandhi** [comment]
> Comments:
> 
> 1. The `data` and `analytics` prefixes and `athena_data` and `athena_analytics` Glue DB names seem pretty arbitrary (could these be parametrized with defaults remaining as above?)
> 2. Would the `table` parameter get confusing when you are doing JOIN across multiple tables?
> 3. Would there ever be a need to trigger the Athena Stage as soon as new data comes in to `athena_data` rather than using cronExpressions? Is that possible to have glue data catalog changes as a listener?

As discussed, This example showcases some of the functionalities of the DDK not covered in other examples, a user/developer can customize it as per their requirements for 1. and 3. For 2. i have taken note and will push another commit to simplify it

File: athena-query-execution-pipeline/README.md [hunk 1]
```diff
@@ -0,0 +1,161 @@
+# Serverless Athena Query Execution Data Pipeline
+
+## Overview
+This pattern deploys serverless data pipelines that handles Athena Query Execution of SQL queries as part of a step function pipeline on a scheduled event pattern which is followed by a glue transform using glue job and crawlers as part of a step function pipeline using a event driven pattern. The code leverages the [AWS DataOps Development Kit (DDK)](https://awslabs.github.io/aws-ddk/) to deploy the infrastructure.
+
+
+## Architecture
```
Comment (malachi-constant) [inline athena-query-execution-pipeline/README.md:7]: Diagram looks great, glad to see a handful of stages incorporated!

File: athena-query-execution-pipeline/athena_query_execution/athena_query_execution_pipeline.py [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+        # Sqs Lambda Stage
+        # ------------------------------------#
+        sqs_lambda_stage = SqsToLambdaStage(
+            self,
+            id="sqs-lambda-stage",
+            environment_id=environment_id,
+            code=lambda_code.from_asset("./athena_query_execution/lambda_handlers"),
+            handler="handler.lambda_handler",
+            layers=[pandas_sdk_layer(self)],
```
Comment (malachi-constant) [inline athena-query-execution-pipeline/athena_query_execution/athena_query_execution_pipeline.py:72]: Yay!

File: athena-query-execution-pipeline/athena_query_execution/configs.json [hunk 3]
```diff
@@ -0,0 +1,13 @@
+{
```
Comment (malachi-constant) [inline athena-query-execution-pipeline/athena_query_execution/configs.json:1]: Is there a reason we don't leverage `ddk.json` here?

File: athena-query-execution-pipeline/requirements.txt [hunk 4]
```diff
@@ -0,0 +1,6 @@
+aws-cdk-lib==2.54.0
```
Comment (malachi-constant) [inline athena-query-execution-pipeline/requirements.txt:1]: Do you want to just bump this to `2.61.0` while we're at it?

**Review by malachi-constant — APPROVED** [review]
Looks awesome @anmolsgandhi, this is quite valuable IMO. Left a few minor comments but nothing critical.

File: athena-query-execution-pipeline/athena_query_execution/configs.json (hunk 3 above)
Comment (anmolsgandhi) [reply athena-query-execution-pipeline/athena_query_execution/configs.json:1]: I wanted to keep ddk infra config separate from usecase config, The config file can have 100s of queries for each env which could make the ddk.json file too crowded.

File: athena-query-execution-pipeline/requirements.txt (hunk 4 above)
Comment (anmolsgandhi) [reply athena-query-execution-pipeline/requirements.txt]: Done !

**Review by noah-paige — APPROVED** [review]
looks good! @anmolsgandhi

File: athena-query-execution-pipeline/athena_query_execution/configs.json (hunk 3 above)
Comment (malachi-constant) [reply athena-query-execution-pipeline/athena_query_execution/configs.json:1]: Understood, fine by me

**Review by kukushking — APPROVED** [review]
This is awesome! I think it could be incredibly useful for running [CTAS](https://docs.aws.amazon.com/athena/latest/ug/create-table-as.html) queries to create/update sort of "materialised views" on schedule - a pattern that I've seen with some customers. I'm not sure if it makes sense to also create a separate example featuring just the queries bit and keep this as an example of a more complex pipeline but open to suggestions. Very cool!

**Pull request merged**